### PR TITLE
Feature/docs/281 add documentation on advanced installation and configuration

### PR DESCRIPTION
**Pull request by RabbITCybErSeC**

### CODE REVIEW

File: docs/content/en/docs/installation-configuration/authentication.md [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+
+SOARCA leverages our internally developed [gauth library](https://github.com/COSSAS/gauth) as its underlying authentication framework. This library provides convenient Role-Based Access Control (RBAC) middleware, which manages authentication for various endpoints, such as the Playbook API. Based on OpenID Connect (OIDC), the library supports integration with multiple authentication providers.
+
+{{% alert title="Warning" color="warning" %}}
+Currently, [gauth library](https://github.com/COSSAS/gauth) is supported and tested authentication for the [Authentik](https://goauthentik.io/) authentication provider, an open-source solution that supports a wide range of authentication methods. As such, other OIDC-based providers might not be compatible.
+{{% /alert %}}
+
+{{% alert title="Note" color="primary" %}}
+For the current version, users/application must belong to the `soarca_admin` group to access the API's when authentication is enabled. Furthermore, currently there is not yet control over what user can go to which route. This feature is scheduled for a future version of SOARCA.
+{{% /alert %}}
```
Comment (MaartendeKruijf) [inline docs/content/en/docs/installation-configuration/authentication.md:22]: This is confusing make the group configurable and make a list of supported providers

File: docs/content/en/docs/installation-configuration/authentication.md [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+{{< /tabpane >}}
+
+## Setting up Authentik with SOARCA
+
+Next, we need to obtain variables such as `OIDC_ISSUER` etc. This section will describe how we can setup [Authentik](https://goauthentik.io/).
+
+### Spinning up Authentik
+
+Instruction and docker-compose on how to bundle SOARCA with Authentik will come!
+
```
Comment (MaartendeKruijf) [inline docs/content/en/docs/installation-configuration/authentication.md:45]: Either point to a docker compose or provide one in deployment

File: docs/content/en/docs/installation-configuration/authentication.md [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+### Making an authentication provider
+
+In Authentik first setup a provider. An example configuration is given below:
+
+![core](/SOARCA/images/installation_configuration/authentik_setup/setup-provider.png)
```
Comment (MaartendeKruijf) [inline docs/content/en/docs/installation-configuration/authentication.md:50]: Make sure all pictures a in light mode

File: docs/content/en/docs/installation-configuration/authentication.md [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+OIDC_SKIP_TLS_VERIFY: true
+{{< /tab >}}
+
+{{< tab header="`bash`" lang="bash" >}}
+export AUTH_ENABLED=true
+export OIDC_ISSUER="https://authentikuri:9443/application/o/does-providing-for-soarca/"
+export OIDC_CLIENT_ID="WxUcBMGZdI7c0e5oYp6mYdEd64acpXSuWKh8zBH5"
+export OIDC_SKIP_TLS_VERIFY=true
+{{< /tab >}}
+{{< /tabpane >}}
```
Comment (MaartendeKruijf) [inline docs/content/en/docs/installation-configuration/authentication.md:104]: we could use this style of init which I like better then the one we use now:
see: https://docs.opencti.io/latest/deployment/installation/#configure-the-environment
```
(cat << EOF
OPENCTI_ADMIN_EMAIL=admin@opencti.io
OPENCTI_ADMIN_PASSWORD=ChangeMePlease
OPENCTI_ADMIN_TOKEN=$(cat /proc/sys/kernel/random/uuid)
OPENCTI_BASE_URL=http://localhost:8080
MINIO_ROOT_USER=$(cat /proc/sys/kernel/random/uuid)
MINIO_ROOT_PASSWORD=$(cat /proc/sys/kernel/random/uuid)
RABBITMQ_DEFAULT_USER=guest
RABBITMQ_DEFAULT_PASS=guest
ELASTIC_MEMORY_SIZE=4G
CONNECTOR_HISTORY_ID=$(cat /proc/sys/kernel/random/uuid)
CONNECTOR_EXPORT_FILE_STIX_ID=$(cat /proc/sys/kernel/random/uuid)
CONNECTOR_EXPORT_FILE_CSV_ID=$(cat /proc/sys/kernel/random/uuid)
CONNECTOR_IMPORT_FILE_STIX_ID=$(cat /proc/sys/kernel/random/uuid)
CONNECTOR_EXPORT_FILE_TXT_ID=$(cat /proc/sys/kernel/random/uuid)
CONNECTOR_IMPORT_DOCUMENT_ID=$(cat /proc/sys/kernel/random/uuid)
SMTP_HOSTNAME=localhost
EOF
) > .env
```

and 

`export $(cat .env | grep -v "#" | xargs)`

File: docs/content/en/docs/installation-configuration/authentication.md [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+
+An example curl command is provided below:
+
+{{< tabpane langEqualsHeader=false  >}}
+{{< tab header="`bash`" lang="bash" >}}
+curl -X POST "http://localhost:8080/trigger/playbook/" \
+-H "Content-Type: application/json" \
+-H "Authorization: Bearer <replace token>" \
+-d '{
+    "type": "playbook",
```
Comment (MaartendeKruijf) [inline docs/content/en/docs/installation-configuration/authentication.md:156]: Just put the playbook in examples and use:

curl -X POST -H "Content-Type: application/json" -H "Authorization: Bearer <replace token>"  -d @./example/ssh-playbook.json localhost:8080/trigger/playbook

File: examples/docs/auth/go.mod [hunk 6]
```diff
@@ -0,0 +1,3 @@
+module auth
+
+go 1.23.4
```
Comment (MaartendeKruijf) [inline examples/docs/auth/go.mod:3]: This must not be here we did develop a separate lib for this on purpose

File: examples/docs/auth/main.go
Comment (MaartendeKruijf) [inline examples/docs/auth/main.go:1]: remove

File: examples/docs/auth/main.go [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+	value := os.Getenv(key)
+	switch strings.ToLower(value) {
+	case "true", "1", "yes":
+		return true
+	case "false", "0", "no":
+		return false
+	default:
+		return defaultValue
+	}
+}
```
Comment (MaartendeKruijf) [inline examples/docs/auth/main.go:62]: This is the implementation of parsebool so just use the build-in one

File: examples/docs/auth/readme.adoc [hunk 8]
```diff
@@ -0,0 +1,56 @@
+= Authentik OAuth Token Retrieval Client
```
Comment (MaartendeKruijf) [inline examples/docs/auth/readme.adoc:1]: Move to the gauth lib

**Review by MaartendeKruijf — CHANGES_REQUESTED** [review]
Move the RBAC setup to a separate PR

File: docs/content/en/docs/installation-configuration/authentication.md (hunk 2 above)
Comment (RabbITCybErSeC) [reply docs/content/en/docs/installation-configuration/authentication.md:44]: I will add this in a different pull request for now. ok?

File: docs/content/en/docs/getting-started/_index.md [hunk 9]
```diff
@@ -98,11 +97,27 @@ LOG_FORMAT: "json"
...
+THEHIVE_API_TOKEN: your_token
+THEHIVE_API_BASE_URL: http://your.thehive.instance/api/v1/
+
+# Authentication
+
+AUTH_ENABLED: false  
+OIDC_ISSUER: "https://authentikuri:9443/application/o/soarca/"
+OIDC_CLIENT_ID: "some client ID"
+OIDC_SKIP_TLS_VERIFY: false
 {{< /tab >}}
```
Comment (MaartendeKruijf) [inline docs/content/en/docs/getting-started/_index.md:117]: Hive should move aswell

File: docs/content/en/docs/concepts/executable-playbooks.md [hunk 10]
```diff
@@ -12,7 +12,7 @@ resources:
 SOARCA is build on top of the [CACAO Security Playbook Version 2.0](https://docs.oasis-open.org/cacao/security-playbooks/v2.0/cs01/security-playbooks-v2.0-cs01.html) standard.
 
 {{% alert title="Warning" color="warning" %}}
-SOARCA v1.0 only implements a part of the CACAO v2 spec. Only `start`, `end`, and `action` steps are supported at this time.
+SOARCA v1.0 only implements a part of the CACAO v2 spec. Only `start`, `end`, `action`, and `if/else` steps are supported at this time.
```
[the remark this answers is not in the document]
Comment (RabbITCybErSeC) [reply docs/content/en/docs/concepts/executable-playbooks.md:15]: @MaartendeKruijf  what is correct then?

**Pull request merged**